tests/gold/test_when_2d_array_assign.mlir

```diff
@@ -15,6 +15,23 @@ module attributes {circt.loweringOptions = "locationInfoStyle=none"} {
         %3 = hw.array_get %I[%2] : !hw.array<2xi2>, i1
         %4 = hw.constant 1 : i1
         %5 = hw.array_get %I[%4] : !hw.array<2xi2>, i1
-        hw.output %0 : !hw.array<2xi2>
+        %8 = hw.array_get %0[%2] : !hw.array<2xi2>, i1
```

Comment:
is the `emit_when_asserts` option on by default? I don't see where tests are turning it on

Reply:
It is an argument to `m.compile`, e.g. https://github.com/phanrahan/magma/pull/1280/files#diff-b516e2f8f40209812d4f740191a456abcfc72d567ce20bd41720437cb92b793dR40

Reply:
got it...but where in this PR do we pass it to `m.compile()` as `True`? just not seeing it here

Reply:
oops my bad, i didn't realize `tests/test_when.py` was folded so ctrl-f didn't see it. got it :)